pymongo/pool.py

```diff
@@ -1408,7 +1410,10 @@ def get_socket(self, handler=None):
         if self.enabled_for_cmap:
             listeners.publish_connection_check_out_started(self.address)
 
-        sock_info = self._get_socket()
+        sock_info = self._get_socket(error_handler=handler)
+        if handler:
+            handler.contribute_socket(sock_info)
```

Comment:
We don't need this call since contribute_socket is already called directly after the yield (in the calling code).